### ISSUE TITLE
Doctest failures on Gentoo with 9.2.beta9

### ISSUE THREAD

**Issue by strogdon** [cschwan/sage-on-gentoo#595]
Just a FYI. I wanted to make sure I'm on the same page. I no longer have access to a Prefix until someone restores power to the machine. Failures:
```
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/doctest/test.py  # 1 doctest failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/rings/integer.pyx  # 2 doctests failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/combinat/words/paths.py  # 1 doctest failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/geometry/polyhedron/plot.py  # 1 doctest failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/cpython/dict_del_by_value.pyx  # 2 doctests failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/geometry/polyhedron/ppl_lattice_polygon.py  # 1 doctest failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/interfaces/sagespawn.pyx  # 2 doctests failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/interfaces/tests.py  # 1 doctest failed
sage -t --long --warn-long 162.8 --random-seed=0 /usr/lib/python3.7/site-packages/sage/libs/ecl.pyx  # Timed out (and interrupt failed)
```
The `test.py` and `integer.pyx` failures are memory allocation errors and I see then with vanilla sage. The `ecl.pyx` test passed when run individually. The `paths.py, plot.py` and `ppl_lattice_polygon.py` failures seem to behave as on vanilla but with an additional warning
```
sage.repl.rich_output.display_manager.RichReprWarning: Exception in _rich_repr_ while displaying object: The first element of 'code' must be equal to 'MOVETO' (1).  Your first code is 79
```
The `dict_del_by_value.pyx` test fails with
```
      File "sage/cpython/dict_del_by_value.pyx", line 419, in sage.cpython.dict_del_by_value.test_del_dictitem_by_exact_value (/var/tmp/portage/sci-mathematics/sage-9999/work/sage-9999/src-python3_7/build/cythonized/sage/cpython/dict_del_by_value.c:2428)
        del_dictitem_by_exact_value(<PyDictObject *>D, <PyObject *>value, h)
    TypeError: an integer is required
```
and the `sagespawn.pyx` test fails with
```
Failed example:
    E.read().decode('ascii')
Expected:
    u'hello world\r\n'
Got:
    'orld\r\n'
```

**Comment by kiwifb** [comment]
At the moment only `sagespawn.pyx` and `dict_del_by_value.pyx` are supposed to have failures. The first is caused by pexpect 4.8. The second by a missing cython patch. The rest is a bit more concerning. I guess there is no helping the memory issue.

**Comment by strogdon** [comment]
I think I can adjust memory parameters to remedy the memory issue, but not really a solution. So you don't have the bit with
```
sage.repl.rich_output.display_manager.RichReprWarning: Exception in _rich_repr_ while displaying object:
```
I wonder what is the source? I saw this with `9.2.beta8` but things have not changed much since then.

**Comment by kiwifb** [comment]
Version of Matplotlib used?

**Comment by strogdon** [comment]
I was just thinking that! And maybe useflags?
```
[I] dev-python/matplotlib
     Available versions:  ~1.5.3-r2^t[1] ~2.0.0_beta0^t[1] ~2.0.0^t[1] ~2.1.0^t[1] 3.2.2^t (~)3.3.0^t {cairo doc examples excel fltk gtk2 gtk3 latex pyside qt4 qt5 test tk wxwidgets PYTHON_TARGETS="python2_7 python3_6 python3_7 python3_8 python3_9"}
     Installed versions:  3.3.0^t(07:38:18 PM 08/10/2020)(cairo qt5 wxwidgets -doc -examples -excel -gtk3 -latex -test -tk PYTHON_TARGETS="python3_7 -python3_6 -python3_8 -python3_9")
     Homepage:            https://matplotlib.org/
     Description:         Pure python plotting library with matlab like syntax

[1] "local-overlay" /usr/local/portage
```

**Comment by kiwifb** [comment]
Pure 3.3.0 could cause trouble compared to 3.3.1 (sync your tree) but I don't remember them being that kind of error messages.

**Comment by strogdon** [comment]
`3.3.1` fixed things. Vanilla still uses `3.3.0` but it is patched for this very thing (https://trac.sagemath.org/ticket/30176). So, should `=dev-python/matplotlib-3.3*` be changed in the -9999 ebuild? Or does the `*` pull in the latest?

**Comment by kiwifb** [comment]
I probably should put a >=3.3.1. The `3.3*` would check if any 3.3.x is installed and if 3.3.0 is there that's enough. If no 3.3.x is installed then portage will go with the highest 3.3.x keyworded.

**Comment by strogdon** [comment]
I'll close. This was just so I was on the correct page. I'll comment on the memlimit failures on sage-release.

**Comment by kiwifb** [comment]
On that note I have two new doctests failures with the last head from the vbraun branch. But I walked into the maxima upgrade ticket unwarned, so that should be solved by moving to maxima-5.44.0.
